pkgs/development/python-modules/mashumaro/default.nix

```diff
@@ -0,0 +1,69 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, pythonOlder
+, setuptools
+, typing-extensions
+, orjson
+, msgpack
+, pyyaml
+, tomli-w
+, tomli
+, pytestCheckHook
+, ciso8601
+, pendulum
+, pytest-mock
+}:
+
+buildPythonPackage rec {
+  pname = "mashumaro";
+  version = "3.8.1";
+  format = "pyproject";
```

Comment:
I wasn't sure about this so I did some research. TLDR: I still think `pyproject` format is appropriate, and I'll explain (but I can be convinced otherwise).

[PEP 518](https://peps.python.org/pep-0518/#build-system-table) defines the `build-system-table` section and specifies that, when it's missing, it should default to:

```
[build-system]
# Minimum requirements for the build system to execute.
requires = ["setuptools", "wheel"]  # PEP 508 specifications.
```

[PEP 517](https://peps.python.org/pep-0517/) introduces the `build-backend` key that is specified in the setuptools documentation. It says:

> If the pyproject.toml file is absent, or the build-backend key is missing, the source tree is not using this specification, and tools should revert to the legacy behaviour of running setup.py (either directly, or by implicitly invoking the setuptools.build_meta:__legacy__ backend).

What I interpret from this is that all of the tools we use in the Python ecosystem are already required to fall back to using setuptools and setup.py when build-backend is missing.

Given this, I feel like the presence of pyproject.toml should be enough to trigger switching the format. Then, if upstream incorporates PEP 517, it will happen transparently.

Reply:
Certainly, there is a fallback, and this behavior is specified in the PEP, so it seems stable.
I do not have the material to convince you of this, as I do not have enough experience.

@mweinelt seemed to think we should use setuptools, so maybe he can give us his opinion.
https://github.com/nix-community/nix-init/issues/119